### PR TITLE
[WIP] adding a katello plugin generator

### PR DESCRIPTION
**Pull request by thomasmckay**
- [ ] do not copy Gemfile.lock
- [ ] make gemspec generic
- [ ] Should 'grunt copy' checkout sample plugin if not available?

### CODE REVIEW

**Comment by thomasmckay** [comment]
refs: https://github.com/thomasmckay/sample-bastion/pull/3

File: Gruntfile.js [hunk 1]
```diff
@@ -19,9 +19,53 @@ module.exports = function (grunt) {
...
+                            .replace(/sample model/g,"<%= modelSentence %>")
+                            .replace(/sampleModel/g,"<%= modelCamel %>");
+                    }
+                }
+            }
         }
     });
 
-    grunt.registerTask('default', ['eslint', 'mochaTest']);
+    grunt.registerTask('default', ['eslint', 'mochaTest', 'updatePlugin']);
```
Comment (ehelms) [inline Gruntfile.js:69]: I haven't tried this but I think it won't call what you expect

File: generators/plugin/templates/Gemfile.lock [hunk 2]
```diff
@@ -0,0 +1,86 @@
+PATH
```
Comment (ehelms) [inline generators/plugin/templates/Gemfile.lock:1]: Why include the lock file? Seems hard to maintain

File: generators/plugin/templates/dot_gitignore [hunk 3]
```diff
@@ -0,0 +1,37 @@
+*.rbc
```
Comment (ehelms) [inline generators/plugin/templates/dot_gitignore:1]: You could just call this file gitignore I think most would know it becomes a .gitignore file

File: generators/plugin/templates/sample_bastion_plugin.gemspec [hunk 4]
```diff
@@ -0,0 +1,20 @@
+$:.push File.expand_path("../lib", __FILE__)
+
+# Maintain your gem's version:
+require "<%= nameSnake %>/version"
+
+# Describe your gem and declare its dependencies:
+Gem::Specification.new do |s|
+  s.name        = "<%= nameSnake %>"
+  s.version     = <%= namePascal %>::VERSION
+  s.authors     = ["thomasfmckay@gmail.com"]
```
Comment (ehelms) [inline generators/plugin/templates/sample_bastion_plugin.gemspec:10]: Default?

File: Gruntfile.js [hunk 5]
```diff
@@ -19,9 +19,53 @@ module.exports = function (grunt) {
                 },
                 src: ['test/**/*.test.js']
             }
+        },
+        copy: {
```
Comment (ehelms) [inline Gruntfile.js:23]: Not as obvious to me what this is doing

File: Gruntfile.js (hunk 1 above)
Comment (thomasmckay) [inline Gruntfile.js:69]: I tried naming the target but didn't figure out how. I could keep it as "copy" but wanted to make it specific to the plugin. I'll keep trying.

File: generators/plugin/templates/Gemfile.lock (hunk 2 above)
Comment (thomasmckay) [inline generators/plugin/templates/Gemfile.lock:1]: Good point. I haven't tried excluding files yet but will.

File: generators/plugin/templates/dot_gitignore (hunk 3 above)
Comment (thomasmckay) [inline generators/plugin/templates/dot_gitignore:1]: I wanted a consistent syntax so i could make a generic rule for yeoman. I may revisit it but prefer something to distinguish the "dot" templates.

File: generators/plugin/templates/sample_bastion_plugin.gemspec (hunk 4 above)
Comment (thomasmckay) [inline generators/plugin/templates/sample_bastion_plugin.gemspec:10]: Noted. I will make the final gemspec be either fully configurable via yo or leave as defaults that the plugin creator will need to fill in.

File: Gruntfile.js (hunk 5 above)
Comment (thomasmckay) [inline Gruntfile.js:23]: This is the target that copies the sample plugin and converts it to yeoman templates.

Comment (ehelms) [inline Gruntfile.js:23]: So you have to have the sample plugin locally for this to work? Should this tackle acquiring it for you?

Comment (thomasmckay) [inline Gruntfile.js:23]: Yes, a local copy is expected. Let's revisit that once this PR is closer to being finished and we can run it live to see what feels best.

**Comment by ehelms** [comment]
@thomasmckay any plan to revisit this anytime soon?

**Pull request closed without merging**